### PR TITLE
shields: adafruit_2_8_tft_touch_v2: add arduino nano connector support

### PR DESCRIPTION
**Pull request by jfischer-no**
Allow shield to be used with any board that has an
arduino nano connector.

### CODE REVIEW

File: boards/shields/adafruit_2_8_tft_touch_v2/adafruit_2_8_tft_touch_v2_nano.overlay [hunk 1]
```diff
@@ -14,9 +16,9 @@
 		#gpio-cells = <2>;
 		gpio-map-mask = <0xffffffff 0xffffffc0>;
 		gpio-map-pass-thru = <0 0x3f>;
-		gpio-map = <10 0 &arduino_nano_header 4 0>,  // D4
-			   <15 0 &arduino_nano_header 10 0>, // D10
-			   <16 0 &arduino_nano_header 9 0>;  // D9
+		gpio-map = <10 0 &arduino_nano_header 4 0>,  /* D4  */
+			   <15 0 &arduino_nano_header 10 0>, /* D10 */
+			   <16 0 &arduino_nano_header 9 0>;  /* D9  */
```
Comment (brgl) [inline boards/shields/adafruit_2_8_tft_touch_v2/adafruit_2_8_tft_touch_v2_nano.overlay:21]: I understand the logic behind this commit but I wonder how stable that HW interface is? It looks like the pins on the nano header can be used differently depending on the board. For instance: looking at the Nano IoT DT I'm seeing that the RX and TX pins seem to be inverted compared to BLE? Are we sure those GPIOs that are mapped here will be the same on all boards?

**Comment by brgl** [comment]
@jfischer-no gentle ping on my question

**Comment by jfischer-no** [comment]
> @jfischer-no gentle ping on my question

What question or where?

**Comment by brgl** [comment]
image

@jfischer-no are you not seeing this review?

**Comment by jfischer-no** [comment]
>@jfischer-no are you not seeing this review?

Nope, and according to your screenshot, it is still pending.

**Review by brgl — CHANGES_REQUESTED** [review]

**Comment by brgl** [comment]
> > @jfischer-no are you not seeing this review?
> 
> Nope, and according to your screenshot, it is still pending.

Duh... Sorry for that -ETOOTIRED

File: boards/shields/adafruit_2_8_tft_touch_v2/adafruit_2_8_tft_touch_v2_nano.overlay (hunk 1 above)
Comment (jfischer-no) [reply boards/shields/adafruit_2_8_tft_touch_v2/adafruit_2_8_tft_touch_v2_nano.overlay:21]: > looking at the Nano IoT DT I'm seeing that the RX and TX pins seem to be inverted compared to BLE? Are we sure those GPIOs that are mapped here will be the same on all boards?

That is the reason to have arduino_nano_header (same as arduino_header), it must be stable, if these pins are different then it is a problem with the board, or it does not have an arduino_nano connector.

Comment (jfischer-no) [reply boards/shields/adafruit_2_8_tft_touch_v2/adafruit_2_8_tft_touch_v2_nano.overlay:21]: NANO IOT
NANO BLE

Comment (brgl) [reply boards/shields/adafruit_2_8_tft_touch_v2/adafruit_2_8_tft_touch_v2_nano.overlay:21]: So I take it that the following snippet from `boards/arm/arduino_nano_33_iot/arduino_nano_r3_connector.dtsi` is a bug?

```
 12                 gpio-map = <0 0 &portb 23 0>,  /* D0 / UART-RX */
 13                            <1 0 &portb 22 0>,  /* D1 / UART-TX */
```

Comment (brgl) [reply boards/shields/adafruit_2_8_tft_touch_v2/adafruit_2_8_tft_touch_v2_nano.overlay:21]: Ok this isn't relevant for this PR anyway, I'll open up a different thread.

**Pull request merged**